src/compiler/parser.ts

```diff
@@ -3210,6 +3285,155 @@ module ts {
             node.name = parseRightSideOfDot(/*allowIdentifierNames*/ true);
             return finishNode(node);
         }
+        
+        // Return a string representation of an EntityName
+        function entityNameToString(entityName: EntityName): string {
+            if (entityName.kind === SyntaxKind.Identifier) {
+                return (<Identifier>entityName).text;
+            } else {
+                let qualifiedName = <QualifiedName>entityName;
+                return entityNameToString(qualifiedName.left) + '.' + entityNameToString(qualifiedName.right);
+            }
+            return "";
+        }
+
+        function parseJsxChild(): JsxText | JsxExpression | JsxElement {
+            let result: JsxText | JsxExpression | JsxElement = undefined;
+            switch (token) {
+                case SyntaxKind.JsxText:
+                    result = <JsxText>createNode(SyntaxKind.JsxText);
+                    token = scanner.scanJsxToken();
+                    result = finishNode(result);
+                    break;
+                case SyntaxKind.OpenBraceToken:
+                    result = parseJsxExpression();
+                    break;
+                default:
+                    Debug.assert(token === SyntaxKind.LessThanToken);

```

Comment:
I usually prefer this way so that functions you really care about can get to 100% code coverage, but I'm open to more discussion on this one.